### PR TITLE
LPD-23140 Use snapshot to make it available during upgrade process

### PR DESCRIPTION
**Pull request by devpedrotavares**
Related ticket: [LPD-23140](https://liferay.atlassian.net/browse/LPD-23140)

### CODE REVIEW

**Review by carolmariaabb — APPROVED** [review]

**Comment by carolmariaabb** [comment]
ci:test:relevant

**Comment by carolmariaabb** [comment]
ci:test:forms

**Comment by devpedrotavares** [comment]
ci:report:2560485050

**Comment by devpedrotavares** [comment]
ci:report:2560489298

**Comment by devpedrotavares** [comment]
ci:report:2560489298

**Comment by devpedrotavares** [comment]
ci:test:sf

**Comment by devpedrotavares** [comment]
Hi @carolmariaabb , I added a commit fixing the tests that are failing

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-data-provider-impl/src/test/java/com/liferay/dynamic/data/mapping/data/provider/internal/DDMDataProviderInvokerImplTest.java [hunk 1]
```diff
@@ -254,8 +274,17 @@ public void testFetchDDMDataProviderInstance2() throws Exception {
...
+
+		Mockito.when(
+			snapshot.get()
+		).thenReturn(
+			ddmDataProviderInstanceService
+		);
+
+		ReflectionTestUtil.setFieldValue(
+			ddmDataProviderInvokerImpl,
+			"ddmDataProviderInstanceServiceSnapshot", snapshot);
```
Comment (carolmariaabb) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-data-provider-impl/src/test/java/com/liferay/dynamic/data/mapping/data/provider/internal/DDMDataProviderInvokerImplTest.java:287]: @devpedrotavares move the `DDMDataProviderInvokerImpl ddmDataProviderInvokerImpl =` like you did for the other two places.

**Review by carolmariaabb — CHANGES_REQUESTED** [review]

**Review by carolmariaabb — APPROVED** [review]

**Comment by carolmariaabb** [comment]
ci:forward

**Pull request closed without merging**